shell/platform/windows/flutter_windows.cc

```diff
@@ -188,6 +188,20 @@ HWND FlutterDesktopViewGetHWND(FlutterDesktopViewRef view) {
   return ViewFromHandle(view)->GetPlatformWindow();
 }
 
+bool FlutterDesktopViewGetGraphicsAdapter(FlutterDesktopViewRef view,
+                                          IDXGIAdapter** adapter) {
+  auto surface_manager = ViewFromHandle(view)->GetEngine()->surface_manager();
+  if (surface_manager) {
+    Microsoft::WRL::ComPtr<ID3D11Device> d3d_device;
+    Microsoft::WRL::ComPtr<IDXGIDevice> dxgi_device;
+    if (surface_manager->GetDevice(d3d_device.GetAddressOf()) &&
+        SUCCEEDED(d3d_device.As(&dxgi_device))) {
+      return SUCCEEDED(dxgi_device->GetAdapter(adapter));
+    }
+  }
+  return false;
+}
```

Comment:
Rather than return both a bool and the `IDXGIAdapter*` via an out param, just return `IDXGIAdapter*`. If any part of the process fails, return `nullptr`.